frontend/src/console/components/Hero/Hero.tsx

```diff
@@ -0,0 +1,287 @@
+import * as React from 'react';
+import { InstancesModel } from 'console/models';
+import { StaticContentOverrides, ThemeSchema } from 'ocim-client';
+import {
+  ConsolePage,
+  ConsolePageCustomizationContainer
+} from 'console/components';
+import { WrappedMessage } from 'utils/intl';
+import { Col, Row } from 'react-bootstrap';
+import {
+  CollapseEditArea,
+  ColorInputField,
+  HeroPreview,
+  ImageUploadField,
+  TextInputField
+} from 'ui/components';
+import {
+  clearErrorMessage,
+  updateImages,
+  updateStaticContentOverridesFieldValue,
+  updateThemeFieldValue
+} from 'console/actions';
+import { RootState } from 'global/state';
+import { connect } from 'react-redux';
+import messages from './displayMessages';
+import './styles.scss';
+
+interface State {
+  [key: string]: string;
+
+  title: string;
+  subtitle: string;
+}
+
+interface ActionProps {
+  clearErrorMessage: Function;
+  updateThemeFieldValue: Function;
+  updateImages: Function;
+  updateStaticContentOverridesFieldValue: Function;
+}
+
+interface StateProps extends InstancesModel {}
+
+interface Props extends StateProps, ActionProps {}
+
+export class HeroComponent extends React.PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      title: '',
+      subtitle: ''
+    };
+
+    if (this.homePageOverlayHtmlExists()) {
```

Comment:
This is `false` on page refresh. That's why it will work only when coming from another page.
We should add a `FIXME` to this part and address it in a follow-up, as this is a bug.

Reply:
@Agrendalath, the issue here is not the `this.activeInstance.data.staticContentOverrides.homepageOverlayHtml` being undefined, but `this.activeInstance.data` itself being undefined which causes this condition to return `false`. This appears to be a component lifecycle issue which is also affecting a few other pages, if not all of them.

I added a `componentDidUpdate` method similar to the one in the `InstanceSettings` component and that didn't help. I also tried adding a `getDerivedStateFromProps` method and while that solved this issue, it also made the text input fields read-only for some reason.

I asked @giovannicimolin about this data loading issue and he mentioned that he will create a separate task for investigating and fixing this issue.

Reply:
@Agrendalath, I have come up with a way to fix this data loading issue - will update the PR shortly. However, that has introduced a bug when the user tries to empty either text box. Since I have spent way more time than estimated on this task already, I think we should try to fix this in a follow-up task. Is that okay @giovannicimolin?

Reply:
Sure! We can leave this for a follow up task. Can you add a note to the epic explaining the issue and how to reproduce it?